### PR TITLE
DataFrameMapper.inverse_transform() for simple transformations

### PR DESCRIPTION
**Pull request by erikjandevries**
I've added an `inverse_transform()` method to the `DataFrameMapper` that works for simple transformations.
I've included tests using the `LabelEncoder` and `LabelBinarizer`, which are passed.

This still fails for more complicated transformations such as `Pipeline`s. I hope it's a useful start at least.

### CODE REVIEW

**Comment by erikjandevries** [comment]
Not sure what's going wrong - when I tested the solution, all tests passed. Should I have tested differently?

```
$ python -m pytest -s -q tests/test_dataframe_mapper.py
/usr/lib/python3.6/site-packages/sklearn/cross_validation.py:41: DeprecationWarning: This module was deprecated in version 0.18 in favor of the model_selection module into which all the refactored classes and functions are moved. Also note that the interface of the new CV iterators are different from that of this module. This module will be removed in 0.20.
  "This module will be removed in 0.20.", DeprecationWarning)
...................................................
============================================================================================================= warnings summary ==============================================================================================================
tests/test_dataframe_mapper.py::test_list_transformers
  /usr/lib/python3.6/site-packages/sklearn/utils/validation.py:444: DataConversionWarning: Data with input dtype int64 was converted to float64 by StandardScaler.
    warnings.warn(msg, DataConversionWarning)

-- Docs: http://doc.pytest.org/en/latest/warnings.html
51 passed, 1 warnings in 4.82 seconds

```

**Comment by devforfu** [comment]
@erikjandevries Click [Details](https://circleci.com/gh/scikit-learn-contrib/sklearn-pandas/52?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) link near CircleCI message to see what is going wrong. Mostly - PEP8 violations, as I can see.

**Comment by erikjandevries** [comment]
@devforfu Thanks for the hints, indeed they were PEP8 violations, which I've now fixed.
I guess in my opinion, some PEP8 rules make my code less readable, but I understand the need for standardisation when working in (larger) teams :)

File: sklearn_pandas/dataframe_mapper.py [hunk 1]
```diff
@@ -328,3 +334,34 @@ def transform(self, X):
...
+
+        X       the data to inverse transform
+        """
+
+        X_inv = pd.DataFrame()
+        # We will populate the inverse transformed dataframe column by column
+
+        # Let's keep track of the column we've processed
+        prev_col = 0
+        for columns, transformers, transformed_cols in self.transformed_cols_:
```
Comment (dukebody) [inline sklearn_pandas/dataframe_mapper.py]: Can be replaced by:
```
for built_feature, transformed_cols in zip(self.built_features, self.transformed_names_):
    transformed_cols = self.get_names(columns, transformers, X, alias)
    columns, transformers, _ = built_feature
```

**Comment by devforfu** [comment]
@erikjandevries Do you think that it is possible to address the issues pointed by @dukebody? Then we can do a final review and merge into `master`.

**Comment by adithyabsk** [comment]
After a failed PR and some fiddling around, I figured out why that new sub-field was necessary. In the case of one-to-many transformers, it is necessary to maintain a label list that preserves the grouping of the columns. (i.e. ['A'_1, 'A_2', 'A_3'] in the case of the label encoder) The field that @dukebody suggested to use only has these columns preserved in a flat structure. I would vote to merge this PR in as it looks good otherwise.

**Comment by grabantot** [comment]
```
mapper = sklearn_pandas.DataFrameMapper([
    ('index', None),
   ...
```
With `None` transforms I get the error:
```
'NoneType' object has no attribute 'inverse_transform'
```
Though it isn't critical at all, the feature is nice and useful and can be merged as is. Just pointing a direction for further improvement.

**Comment by hu-minghao** [comment]
你好，已收到，谢谢。
